### PR TITLE
Make man page date reproducible

### PR DESCRIPTION
**Pull request by bmwiedemann**
Make man page date reproducible

See https://reproducible-builds.org/ for why this is good

CMake already uses `SOURCE_DATE_EPOCH` for `TIMESTAMP`
The SOURCE_DATE_EPOCH variable is documented in
https://reproducible-builds.org/docs/source-date-epoch/

Also use UTC to be independent of timezone.

This is mostly a rebase of #553 that I finally managed to test on 0.9.0 now.

This PR was done while working on [reproducible builds](https://reproducible-builds.org/) for [openSUSE](https://en.opensuse.org/openSUSE:Reproducible_Builds).

### CODE REVIEW

**Review by t-wissmann — APPROVED** [review]
Thanks for fixing it! Sorry that I forgot about your original PR after winterbreeze had been merged.

Am I correct, that now, package maintainers can set `BUILD_DATE` manually to achieve a reproducable build? By the way, for the man pages in released versions, the man page date can also be taken from the first line matching `Release.* on .*` and only for git versions we need to improvise (e.g. take a `TIMESTAMP` or look up the `git log`)

**Comment by bmwiedemann** [comment]
maintainers can set `SOURCE_DATE_EPOCH` and cmake will use that to fill BUILD_DATE.
rpmbuild and Debian's and ArchLinux' package build automatically set `SOURCE_DATE_EPOCH` to the latest package changelog entry date, so it can be reproducible by default there.

For reproducible git builds, you could set it to `git log -1 --format=%ct`

**Pull request merged**
